### PR TITLE
Trap IOException on stat() failures

### PR DESCRIPTION
**Pull request by TranceLove**
## Description
As band-it solution to prevent crash, by trapping IOException that would cause containing SftpTemplate to not return anything (but null).

#### Issue tracker   
Addresses #2779

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [ ] Done  
  
#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 1]
```diff
@@ -470,11 +470,9 @@ public Boolean execute(SFTPClient client) throws IOException {
                       .stat(SshClientUtils.extractRemotePathFrom(path))
                       .getType()
                       .equals(FileMode.Type.DIRECTORY);
-                } catch (SFTPException notFound) {
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:473]: What happens in case of `SFTPException` why did you remove it?

Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:473]: `net.schmizz.sshj.sftp.SFTPException` extends `net.schmizz.sshj.common.SSHException`, which extends `java.io.IOException`.

**Review by VishalNehra — DISMISSED** [review]

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 2]
```diff
@@ -470,11 +470,9 @@ public Boolean execute(SFTPClient client) throws IOException {
...
-                  Log.e(
-                      getClass().getSimpleName(),
-                      "Fail to execute isDirectory for SFTP path :" + path);
-                  notFound.printStackTrace();
+                } catch (IOException executionFailure) {
+                  Log.e(TAG, "Fail to execute isDirectory for SFTP path :" + path);
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:474]: Do `Log.e(TAG, "Fail to execute isDirectory for SFTP path :" + path, executionFailure);`

**Review by EmmanuelMess — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 3]
```diff
@@ -470,11 +470,9 @@ public Boolean execute(SFTPClient client) throws IOException {
                       .stat(SshClientUtils.extractRemotePathFrom(path))
                       .getType()
                       .equals(FileMode.Type.DIRECTORY);
-                } catch (SFTPException notFound) {
-                  Log.e(
-                      getClass().getSimpleName(),
-                      "Fail to execute isDirectory for SFTP path :" + path);
-                  notFound.printStackTrace();
+                } catch (IOException executionFailure) {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:473]: Does this code not throw `SFTPException`?

Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:473]: `SFTPException` is subclass of `IOException`.

**Review by VishalNehra — DISMISSED** [review]

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java (hunk 3 above)
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:473]: > `SFTPException` is subclass of `IOException`.

Thought it wasn't, my mistake.

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 4]
```diff
@@ -470,11 +470,10 @@ public Boolean execute(SFTPClient client) throws IOException {
...
                       .getType()
                       .equals(FileMode.Type.DIRECTORY);
-                } catch (SFTPException notFound) {
+                } catch (IOException executionFailure) {
                   Log.e(
-                      getClass().getSimpleName(),
-                      "Fail to execute isDirectory for SFTP path :" + path);
-                  notFound.printStackTrace();
+                      TAG, "Fail to execute isDirectory for SFTP path :" + path, executionFailure);
+                  executionFailure.printStackTrace();
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:476]: This printStackTrace is redundant

**Review by VishalNehra — APPROVED** [review]

**Review by VishalNehra — APPROVED** [review]
